### ISSUE TITLE
Dice Loss pytorch clarity

### ISSUE THREAD

**Issue by Taariq78** [voxelmorph/voxelmorph#224]
Hello again.  I am attempting to make use of the dice loss function in pytorch.  

Firstly, why is the definition of the loss function under the `Dice` class different from that of which is found in`utils.py`?  The latter makes use of labels, whereas the former does not.  Why is this so?
![dice_in_loss](https://user-images.githubusercontent.com/70844260/95660033-7f28ae00-0b25-11eb-86fb-994589134e8d.PNG)

![dice_in_utils](https://user-images.githubusercontent.com/70844260/95660034-8059db00-0b25-11eb-9947-ac35e33f6a35.PNG)

Secondly, my current progress is as follows: I made some rough segmentations of my region of interest.  They are images of size: 256x256.  Does the 'label' refer to the pixel value of the segmentations?  I ask this because of how the of how the `dice` function is defined (with the use of logical ands).  If so, would this mean that my segmentations should be of a single pixel value in order to make use of the dice function?  Thank you in advance. 

I attached my segmented image below

![segmentation(top)_corresponding_image(bottom)](https://user-images.githubusercontent.com/70844260/95660036-80f27180-0b25-11eb-8864-8e571ac7a45b.PNG)

**Comment by adalca** [comment]
The first is a differentiable approximation to the Dice loss (a la Milletari 2016) -- it assumes a probabilistic (one-hot-like) representation of the segmentation maps. For using Dice as a loss function, we need this representation (so we can have gradients)

The second is a "hard" Dice (real Dice) which assumes a volume with label ids at each voxel. You can use this in final Dice evaluations.
